### PR TITLE
[IMPROVEMENT] Add extra condition to handle replied message preview

### LINKED ISSUE #723: Last message in chat lists differs from the one inside the channel
- Your Rocket.Chat Experimental app version: 1.10.0.6193
- Your Rocket.Chat server version: Official Rocket.chat instance
- Device (or Simulator) you're running with: Huawei Y6 Pro 2017, Android 7.0

**Description**
The app seems to not be making any update of the `last message` in the chat list.

**Screenshots**
Last message in the chat list (#support channel):

lastmessage_chatlist

Last message inside the channel:
lastmessage_channel

### LINKED ISSUE #776: wrong preview in of replied message RoomList View
WhatsApp Image 2019-04-01 at 11 08 10 PM

### PR DESCRIPTION
**Pull request by pranavpandey1998official**
@RocketChat/ReactNative

this PR added message parser to roomItem message preview and made a new util function to parse replied message as It was not rendering as expected

added story test for replied message 

Closes #723 #776 

before
WhatsApp Image 2019-04-29 at 6 30 48 PM

after
WhatsApp Image 2019-04-29 at 6 30 48 PM (1)

### CODE REVIEW

File: app/presentation/RoomItem.js [hunk 1]
```diff
@@ -203,6 +203,11 @@ export default class RoomItem extends React.Component {
 			prefix = `${ lastMessage.u.username }: `;
 		}
 
+		if (lastMessage.msg && lastMessage.attachments && Object.keys(lastMessage.attachments).length > 0) {
+			lastMessage.msg = lastMessage.msg.replace(/^\[ ]\(.+?\)/, '');
+			prefix = prefix.replace(/:/, ` (${ I18n.t('Replied') }):`);
```
Comment (diegolmello) [inline app/presentation/RoomItem.js]: Remove it. It's cleaner without it.

Comment (diegolmello) [reply app/presentation/RoomItem.js]: Also remove `You:` from prefix if `room.t === 'd'`

Comment (pranavpandey1998official) [reply app/presentation/RoomItem.js]: done

File: app/presentation/RoomItem.js [hunk 2]
```diff
@@ -203,6 +203,13 @@ export default class RoomItem extends React.Component {
 			prefix = `${ lastMessage.u.username }: `;
 		}
 
+		if (lastMessage.msg && lastMessage.attachments && Object.keys(lastMessage.attachments).length > 0) {
+			lastMessage.msg = lastMessage.msg.replace(/^\[ ]\(.+?\)/, '');
```
[the remark this answers is not in the document]
Comment (diegolmello) [reply app/presentation/RoomItem.js]: Add this regex to an util function and apply here: https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/containers/message/Markdown.js#L40

**Comment by diegolmello** [comment]
@pranavpandey1998official Please add a detailed description of your PR.

**Comment by pranavpandey1998official** [comment]
done

**Comment by djorkaeffalexandre** [comment]
Not needed anymore.

**Pull request closed without merging**